spec/markdown-preview-spec.js

```diff
@@ -0,0 +1,857 @@
+/*
+ * decaffeinate suggestions:
+ * DS102: Remove unnecessary code created because of implicit returns
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
+const path = require('path')
+const fs = require('fs-plus')
+const temp = require('temp').track()
+const MarkdownPreviewView = require('../lib/markdown-preview-view')
+const { TextEditor } = require('atom')
+const TextMateLanguageMode = new TextEditor().getBuffer().getLanguageMode()
+  .constructor
+
+describe('Markdown Preview', function () {
+  let preview = null
+
+  beforeEach(function () {
+    const fixturesPath = path.join(__dirname, 'fixtures')
+    const tempPath = temp.mkdirSync('atom')
+    fs.copySync(fixturesPath, tempPath)
+    atom.project.setPaths([tempPath])
+
+    jasmine.unspy(TextMateLanguageMode.prototype, 'tokenizeInBackground')
+
+    jasmine.useRealClock()
+    jasmine.attachToDOM(atom.views.getView(atom.workspace))
+
+    waitsForPromise(() => atom.packages.activatePackage('markdown-preview'))
+
+    waitsForPromise(() => atom.packages.activatePackage('language-gfm'))
+
+    runs(() =>
+      spyOn(atom.packages, 'hasActivatedInitialPackages').andReturn(true)
+    )
+  })
+
+  const expectPreviewInSplitPane = function () {
+    waitsFor(() => atom.workspace.getCenter().getPanes().length === 2)
+
+    waitsFor(
+      'markdown preview to be created',
+      () =>
+        (preview = atom.workspace
+          .getCenter()
+          .getPanes()[1]
+          .getActiveItem())
+    )
+
+    runs(function () {
+      expect(preview).toBeInstanceOf(MarkdownPreviewView)
+      expect(preview.getPath()).toBe(
+        atom.workspace.getActivePaneItem().getPath()
+      )
+    })
+  }
+
+  describe('when a preview has not been created for the file', function () {
+    it('displays a markdown preview in a split pane', function () {
+      waitsForPromise(() => atom.workspace.open('subdir/file.markdown'))
+      runs(() =>
+        atom.commands.dispatch(
+          atom.workspace.getActiveTextEditor().getElement(),
+          'markdown-preview:toggle'
+        )
+      )
+      expectPreviewInSplitPane()
+
+      runs(function () {
+        const [editorPane] = atom.workspace.getCenter().getPanes()
+        expect(editorPane.getItems()).toHaveLength(1)
+        expect(editorPane.isActive()).toBe(true)
+      })
+    })
+
+    describe("when the editor's path does not exist", () =>
```

Comment:
For consistency, should we change all these into `() => { }` blocks, including the ones that are currently `function () { }`?

Reply:
I can do that as well, what I've decided not to do is to change all the tests to use `async/await` and Promises: this would be too much work for now